src/Microsoft.OpenApi.Tool/OpenApiService.cs

```diff
@@ -35,12 +35,20 @@ static class OpenApiService
 
             var result = new OpenApiStreamReader(new OpenApiReaderSettings
             {
-                ReferenceResolution = resolveExternal == true ? ReferenceResolutionSetting.ResolveAllReferences : ReferenceResolutionSetting.ResolveLocalReferences,
+                ReferenceResolution = resolveExternal ? ReferenceResolutionSetting.ResolveAllReferences : ReferenceResolutionSetting.ResolveLocalReferences,
                 RuleSet = ValidationRuleSet.GetDefaultRuleSet()
             }
             ).ReadAsync(stream).GetAwaiter().GetResult();
 
             document = result.OpenApiDocument;
+
+            // Check if filter options are provided, then execute
+            if (!string.IsNullOrEmpty(filterByOperationIds))
+            {
+                var predicate = OpenApiFilterService.CreatePredicate(filterByOperationIds);
+                document = OpenApiFilterService.CreateFilteredDocument(document, predicate);
+            }
+
             var context = result.OpenApiDiagnostic;
 
             if (context.Errors.Count != 0)
```

Comment:
```suggestion
            if (context.Errors.Count > 0)
```
I think this better expresses the intention.